helpdesk_mgmt_fieldservice/models/helpdesk_ticket.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (C) 2019 - TODAY, Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class HelpdeskTicket(models.Model):
+    _inherit = 'helpdesk.ticket'
+
+    fsm_order_ids = fields.One2many('fsm.order', 'ticket_id',
+                                    string='Service Orders')
+    fsm_location_id = fields.Many2one('fsm.location', string='FSM Location')
+
+    all_orders_closed = fields.Boolean(compute='_compute_all_closed',
+                                       store=True)
+
+    @api.multi
+    def write(self, vals):
+        if 'stage_id' in vals:
+            ticket_stage = self.env['helpdesk.ticket.stage'].browse(
+                vals.get('stage_id'))
+            if ticket_stage.closed:
+                for ticket in self:
```

Comment:
This would not be better in a constraint to avoid overloading the writes ??, I am not sure of the possible scenarios, it is just a suggestion :)

Reply:
Hello, thank for your suggestion. I searching samples to make this change. If you have anything sample, send to me please:)

Reply:
Change it from a write method to one decorated with the @api.constrains('stage_id')

Reply:
https://github.com/OCA/field-service/blob/c2188cdb54f768adb640e6c2abf717dac08c775e/fieldservice_stage_validation/models/fsm_order.py#L10

This is a sample for you to clarify.  Rather than use the write method every time the record is written, it only performs the operation only when the stage_id field is changed

Reply:
:+1: